rawkv/rawkv.go

```diff
@@ -307,7 +307,7 @@ func (c *Client) ReverseScan(ctx context.Context, startKey, endKey []byte, limit
 		return nil, nil, errors.WithStack(ErrMaxScanLimitExceeded)
 	}
 
-	for len(keys) < limit {
+	for len(keys) < limit && bytes.Compare(startKey, endKey) > 0 {
```

Comment:
Is it necessary to check len(endKey)?

Reply:
No, if endKey is nil or empty slice, bytes.Compare always gt 0 except startKey is nil too.